pinot-common/src/main/java/org/apache/pinot/common/utils/helix/HelixHelper.java

```diff
@@ -250,6 +250,13 @@ public static IdealState getTableIdealState(HelixManager manager, String resourc
     return accessor.getProperty(builder.idealStates(resourceName));
   }
 
+  public static Map<String, Map<String, String>> getSegmentAssignment(String tableNameWithType,
```

Comment:
(minor) Suggest not adding this as a general util because:
1. We don't usually perform existence check in this util class
2. It is not very clear where the assignment from by the method name

It would be more readable to perform the IS null check on the caller side